src/Umbraco.Core/Scoping/Scope.cs

```diff
@@ -34,6 +35,11 @@ internal class Scope : IScope2
         private ICompletable _fscope;
         private IEventDispatcher _eventDispatcher;
 
+        // ReadLocks and WriteLocks if we're the outer most scope it's those owned by the entire chain
+        // If we're a child scope it's those that we have requested.
+        public Dictionary<int, int> ReadLocks;
```

Comment:
If these are public or accessed outside of this class they need to be properties like `public Dictionary<int, int> WriteLocks {get;}` but I would suggest these be internal.

Reply:
You're absolutely right, there's no reason why they can't be internal, and you shouldn't be able to set these dicts to something different once they've been created, so I've made them internal readonly